### PR TITLE
Refactor workflow lifCyle tests

### PR DESCRIPTION
**Pull request by AsmaaELM**
Issue: https://scality.atlassian.net/browse/ZKUI-128

### CODE REVIEW

File: src/react/workflow/__tests__/CreateWorkflow.test.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      expect(createButton).toBeInTheDocument();
+
+      const cancelButton = screen.getByText('Cancel');
+      expect(cancelButton).toBeInTheDocument();
+
+    } catch (e) {
+      console.log('should render a form for create workflow: ', e);
+      throw e;
+    }
+  });
```
Comment (JBWatenbergScality) [inline src/react/workflow/__tests__/CreateWorkflow.test.tsx]: This test is not enough, we want to interact with the rule type and ensure that when changing it the right form is rendered (expiration vs replicaiton)

File: src/react/workflow/__tests__/ExpirationForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      const spinButton = screen.getAllByRole('spinbutton');
+      expect(spinButton[0].getAttribute('type')).toBe('number');
+      expect(spinButton[1].getAttribute('type')).toBe('number');
+      expect(spinButton[2].getAttribute('type')).toBe('number');
+
+    } catch (e) {
+      console.log('should render a form for expiration workflow: ', e);
+      throw e;
+    }
+  });
```
Comment (JBWatenbergScality) [inline src/react/workflow/__tests__/ExpirationForm.test.tsx]: This test is not enough we want to toggle Current version and ensure delete marker toggle get disabled, also we want to check that the form is valid only when all required filled are correctly filled in

File: src/react/workflow/__tests__/ReplicationForm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      expect(screen.getByText('Prefix')).toBeInTheDocument();
+      expect(screen.getByText('Destination')).toBeInTheDocument();
+      expect(screen.getByText('Location Name')).toBeInTheDocument();
+
+    } catch (e) {
+      console.log('should render a form for replication workflow: ', e);
+      throw e;
+    }
+  });
+});
```
Comment (JBWatenbergScality) [inline src/react/workflow/__tests__/ReplicationForm.test.tsx]: Same as above we want to interract with the form and check if it is valid when all required inputs are correctly filled

**Comment by AsmaaELM** [comment]
ping

File: src/react/workflow/__tests__/ExpirationForm.test.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      const removeExpiredToggleState = screen.getByTestId('toggle-action-remove-expired-markers');
+      const removeExpiredEnabled = removeExpiredToggleState.querySelector(
+        'input[placeholder="expireDeleteMarkersTrigger"]',
+      );
+
+      expect(removeExpiredEnabled).toBeDisabled();
+
+      const formValidation = screen.getByTestId('form-expiration');
+      expect(formValidation.textContent).toBe('form-valid');
+
```
Comment (JBWatenbergScality) [inline src/react/workflow/__tests__/ExpirationForm.test.tsx]: Missing case of bucket not versionned

**Review by JBWatenbergScality — APPROVED** [review]
LGTM !

**Comment by AsmaaELM** [comment]
ping

**Pull request merged**